seat.c

```diff
@@ -727,8 +762,11 @@ seat_create(struct cg_server *server)
 	wl_signal_add(&server->backend->events.new_input, &seat->new_input);
 
 	wl_list_init(&seat->drag_icons);
-	seat->new_drag_icon.notify = handle_new_drag_icon;
-	wl_signal_add(&seat->seat->events.new_drag_icon, &seat->new_drag_icon);
+	seat->request_start_drag.notify = handle_request_start_drag;
+	wl_signal_add(&seat->seat->events.request_start_drag,
+			&seat->request_start_drag);
+	seat->start_drag.notify = handle_start_drag;
+	wl_signal_add(&seat->seat->events.start_drag, &seat->start_drag);
```

Comment:
Should this signal handlers be removed when this seat gets destroyed?

Reply:
This is based on @emersion's work. @emersion?

Reply:
Yes. Is this missing in rootston?

Reply:
Yes, it is.

Reply:
Damn, patches welcome